### PR TITLE
Update perf_train_gpu_many.md

### PR DESCRIPTION
**Pull request by remyleone**
# What does this PR do?

I didn't know what ZeRO stand for, therefore I added a link to the relevant material. I think this page could gain from adding many links to external sources of documentation and jargon.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @stevhliu

File: docs/source/en/perf_train_gpu_many.md [hunk 1]
```diff
@@ -56,15 +56,15 @@ impact performance. Here's a breakdown of your options:
 If your model can comfortably fit onto a single GPU, you have two primary options:
 
 1. DDP - Distributed DataParallel
-2. ZeRO - depending on the situation and configuration used, this method may or may not be faster, however, it's worth experimenting with it.
+2. [ZeRO](https://arxiv.org/pdf/1910.02054) - depending on the situation and configuration used, this method may or may not be faster, however, it's worth experimenting with it.
```
Comment (stevhliu) [inline docs/source/en/perf_train_gpu_many.md:59]: It'd be even clearer to spell out the acronym first :)

```suggestion
2. [Zero Redundancy Optimizer (ZeRO)](https://arxiv.org/abs/1910.02054) - depending on the situation and configuration used, this method may or may not be faster, however, it's worth experimenting with it.
```

File: docs/source/en/perf_train_gpu_many.md [hunk 2]
```diff
@@ -56,15 +56,15 @@ impact performance. Here's a breakdown of your options:
...
 
 **Case 2: Your model doesn't fit onto a single GPU:**
 
 If your model is too large for a single GPU, you have several alternatives to consider:
 
 1. PipelineParallel (PP)
-2. ZeRO
-3. TensorParallel (TP)
+2. [ZeRO](https://arxiv.org/pdf/1910.02054)
```
Comment (stevhliu) [inline docs/source/en/perf_train_gpu_many.md:66]:
```suggestion
2. [ZeRO](https://arxiv.org/abs/1910.02054)
```

**Review by stevhliu — APPROVED** [review]
Thanks for making it clearer!

**Pull request merged**